fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java

```diff
@@ -675,28 +683,53 @@ public PlanFragment visitPhysicalHashJoin(
         if (hashJoin.getOtherJoinConjuncts().isEmpty()
                 && (joinType == JoinType.LEFT_ANTI_JOIN || joinType == JoinType.LEFT_SEMI_JOIN)) {
             for (SlotDescriptor leftSlotDescriptor : leftSlotDescriptors) {
+                if (!leftSlotDescriptor.isMaterialized()) {
+                    continue;
+                }
                 SlotReference sf = leftChildOutputMap.get(context.findExprId(leftSlotDescriptor.getId()));
+                if (sf == null) {
+                    continue;
+                }
```

Comment:
add this because we remove project on olapScan? chould we add a method to TupleDescriptor to remove slot from tuple descriptor.
`updateChildSlotsMaterialization` works like remove slot instead of set slot to Materialized.